### PR TITLE
Add new features for an update to v0.1.1

### PR DESCRIPTION
**Pull request by Delgan**
Some months ago, I needed a LaTeX to Unicode converter in JavaScript. I ended up with something mixing [your package](https://github.com/jmeas/latex-to-unicode.js), the @ypsu [Python implementation](https://github.com/ypsu/latex-to-unicode) and the @vikhyat [Ruby implementation](https://github.com/vikhyat/latex-to-unicode).

I would like to share what I did with you in order to improve this repository, for what it's worth.

Basically, I just made some fixes and added some features:
- `\frac{a}{b}`
- `\sqrt[a]{b}`
- `\binom{a}{b}`
- `\accent{a}`

Moreover, I decided to add some mocha tests in order to let you see that I did not add regression, and that your `latex-to-unicode` package still works as expected.

Let me know what you think of this upgrade, and if you want me to change something.

### CODE REVIEW

**Comment by jamesplease** [comment]
Whoa, awesome! I'll take a look at this soon ✌️

File: index.js [hunk 1]
```diff
@@ -9,6 +10,14 @@ var textfrak = require('./data/textfrak');
 var textit = require('./data/textit');
 var textmono = require('./data/textmono');
 
+// Replace some latex symbols with their alias equivalent
+function convertAliases(str) {
+    for (var i = 0; i < aliases.length; i++) {
+        str = str.replace(aliases[i][0], aliases[i][1], 'g');
+    }
+    return str;
+}
```
Comment (jamesplease) [inline index.js:36]: 2 space indentation please :v:

File: data/aliases.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[
+  ["\\mathfrak", "\\frak"],
+  ["\\mathcal", "\\cal"],
+  ["\\mathbb", "\\bb"],
+  ["\\mathbf", "\\bf"],
+  ["\\dfrac", "\\frac"],
+  ["\\ldots", "\\dots"]
+]
```
Comment (jamesplease) [inline data/aliases.json:8]: It's been some time since I've worked with LaTeX, but I remember there being a lot of aliases. Does this open up a can of worms? Also, what source are you using for these?

File: index.js [hunk 3]
```diff
@@ -79,12 +79,72 @@ function applyAllModifiers(str) {
...
+    if (n == '7' && d == '8') frac = '⅞';
+
+    if (frac === '') frac = '(' + n + ' / ' + d + ')';
+  
+    var subLeft = str.substring(0, idx);
+    var subRight = str.substring(idx + key.length + num[1] + den[1]);
+    str = subLeft + frac + subRight;
+  }
+  return str;
+}
```
Comment (jamesplease) [inline index.js:171]: fancy

**Comment by jamesplease** [comment]
Thanks again for the update! This PR really takes this library to the next level. Some feedback, on a per-commit basis:

8db0b917662af1b9ed3c91fb1c94fa9aff7fd96b – not sure if I follow the argument here. If `\\ggg` is replaced with `\\gg`, wouldn't that br an indicator of a bad parser?

I admit it's been some time since I've looked at the LaTeX internals, but I'd expect it to begin recognition of a symbol on a space or a `\`, rather than on a rolling basis, if that makes sense.

42ec9d7ee68d122dd22b55de9e3ca13a93c13a29 - looks good. I'm curious; what source are you using for these?

07b851b6ba3348a30aa90e4d1ec01587a9dfa627 - I can bump the version when I cut a release

File: data/aliases.json (hunk 2 above)
Comment (Delgan) [inline data/aliases.json:8]: I have to admit that I do not know much about LaTeX. Actually, I just translated the Ruby code from @vikhyat, as you see, those are the aliases he used [here](https://github.com/vikhyat/latex-to-unicode/blob/master/lib/data/aliases). 

I just made some researches but I was not able to find much info about aliases in LaTeX. However, I found [a question on StackOverflow](http://stackoverflow.com/questions/28444744/how-do-i-alias-command-mathbb), and an user is asking how to declare `\\bb` as the alias of `\\mathbb`. So, what I understood is that there is not any "built-in" alias in LaTeX, am I wrong? So this file may be used for users whose want to have custom aliases.

**Comment by Delgan** [comment]
> 8db0b91 – not sure if I follow the argument here. If \ggg is replaced with \gg, wouldn't that br an indicator of a bad parser?
> 
> I admit it's been some time since I've looked at the LaTeX internals, but I'd expect it to begin recognition of a symbol on a space or a \, rather than on a rolling basis, if that makes sense.

This is not exactly that `\\ggg` would be replaced with `\\gg`.
Before my commit, symbols was replaced like this:

``` javascript
Object.keys(symbols).forEach(function(key) {
  var val = symbols[key];
  str = str.replace(key, val);
});
```

So the associative array is iterated in unpredictable order, and key are replaced in the string with corresponding symbol step by step (by the way, I cannot see `g` flag, so I am not sure it correctly replaces multiple same symbol).
Suppose that you want to translate `999 \\ggg 1`, as we do not know in which order keys will be iterated, it may happen that `str = str.replace('\\gg', '≫')` is applied before `str = str.replace(' \\ggg', '⋙')`, so the result will be `999 ≫g 1` instead of the expected `999 ⋙ 1`.
This is why I thought the longest keys should be replaced first, to avoid this kind of issue. Do you agree, or did I miss something?

> 42ec9d7 - looks good. I'm curious; what source are you using for these?

I stolen it from [here](https://github.com/vikhyat/latex-to-unicode/blob/master/lib/data/symbols). :smile: 

> 07b851b - I can bump the version when I cut a release

I reverted the last commit.

**Comment by jamesplease** [comment]
> So the associative array is iterated in unpredictable order, and key are replaced in the string with corresponding symbol step by step (by the way, I cannot see g flag, so I am not sure it correctly replaces multiple same symbol).

Ah, of course! Good catch.

Thanks for filling in the details. Gonna review this again soon and get it merged. Thanks again so much for this great work!

**Review by luckymore — APPROVED** [review]
